src/onegov/org/forms/extensions.py

```diff
@@ -298,25 +303,29 @@ class PushNotificationFormExtension(FormExtension[FormT], name='publish'):
     def create(self, timezone: str = 'Europe/Zurich') -> type[FormT]:
 
         class PublicationForm(self.form_class):  # type:ignore
-            send_push_notifications_to_app = BooleanField(
-                label=_('Send push notifications to app'),
-                fieldset=_('Publication'),
-                validators=[StrictOptional()],
-                render_kw={'disabled': 'disabled'},  # Starts as disabled
-            )
-            push_notifications = MultiCheckboxField(
-                label=('Topics'),
-                choices=[],
-                fieldset=_('Publication'),
-                depends_on=('send_push_notifications_to_app', 'y'),
-                validators=[StrictOptional()],
-                render_kw={'class_': 'indent-form-field'}
-            )
+            def __init__(self, *args: Any, **kwargs: Any) -> None:
+                super().__init__(*args, **kwargs)
+                if self.request.app.org.firebase_adminsdk_credential:
+                    self._add_push_notification_fields()
+
+            def _add_push_notification_fields(self) -> None:
+                self.send_push_notifications_to_app = BooleanField(
+                    label=_('Send push notifications to app'),
+                    fieldset=_('Publication'),
+                    validators=[StrictOptional()],
+                    render_kw={'disabled': 'disabled'},
+                )
+                self.push_notifications = MultiCheckboxField(
+                    label=('Topics'),
+                    choices=[],
+                    fieldset=_('Publication'),
+                    depends_on=('send_push_notifications_to_app', 'y'),
+                    validators=[StrictOptional()],
+                    render_kw={'class_': 'indent-form-field'},
+                )
```

Comment:
Generally we go the other way, we always add the fields, and then in `on_request` we delete them, if we don't need them. I would suggest doing it that way for consistency.

Or better yet, in this case you can probably just do this at the top of the function:

```python
if not self.request.app.org.firebase_adminsdk_credential:
    return self.form_class
```

Then you don't need any branches below at all.

Reply:
I think that is what I tried before, this will not work

```python
    def create(self, timezone: str = 'Europe/Zurich') -> type[FormT]:

>       if not self.request.app.org.firebase_adminsdk_credential:
E       AttributeError: 'PushNotificationFormExtension' object has no attribute 'request'
```

Reply:
Oh right, `self` refers to the the form in that  context, not the extension.